tests/build_cache_images.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+set -e
+
+CACHE_REGISTRY=${1}
+APP_NAME=${2}
+TARGET_NAME=${3}
+DOCKERFILE=${4:-"Dockerfile"}
+
+if [ -z "$CACHE_REGISTRY" ] || [ -z "$APP_NAME" ] || [ -z "$TARGET_NAME" ]; then
+    echo "Usage: build-cache-images.sh cache-registry appname target-name [dockerfile-name]"
+    exit 1
+fi
+
+# cd to the directory of this script
+cd "$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")"
+
+# Ensure the app exists
+APP_DIR="apps/${APP_NAME}"
+if [ ! -d "$APP_DIR" ]; then
+    echo "App $APP_NAME does not exist"
+    exit 1
+fi
+if [ ! -e "$APP_DIR/${DOCKERFILE}" ]; then
+    echo "No Dockerfile (${DOCKERFILE}) found in $APP_NAME folder"
+    exit 1
+fi
+
+function hashdir(){
+    starting_dir="$(pwd)"
+    target_dir="$1"
+    cd "$target_dir"
+
+    array=($(find . -not -type d))
+    readarray -t sorted < <(sort < <(printf '%s\n' "${array[@]}"))
+    result=""
+    for i in ${sorted[@]};do
+        result+="$(sha256sum $i)\n"
+    done
+    cd "$starting_dir"
+    echo -e "$result" | sha256sum | awk '{ print $1 }'
+}
+
+# Compute the hash of the app's files
+HASH=$(hashdir "$APP_DIR")
+echo "HASH: ${HASH:0:10} (${HASH})"
+HASH="${HASH:0:10}"
+CACHE_TAG="${CACHE_REGISTRY}/${APP_DIR}:${DOCKERFILE}-${HASH}"
```

Comment:
CACHE_NAME is better, it is not only about tag.

Reply:
Done